kasparovd/controllers/transaction.go

```diff
@@ -114,6 +114,37 @@ func GetTransactionsByAddressHandler(address string, skip, limit int64) (interfa
 	}, nil
 }
 
+// GetTransactionsByBlockHashHandler retrieves all transactions
+// included by the block with the given blockHash.
+func GetTransactionsByBlockHashHandler(blockHash string) (interface{}, error) {
+	preloadedFields := dbmodels.PrefixFieldNames(dbmodels.BlockFieldNames.Transactions, dbmodels.TransactionRecommendedPreloadedFields)
```

Comment:
Are we sure that if we preload block.Transactions we get it in Index order?
My guess is that it will probably be that way because this is the order stuff is inserted into the database, but the SQL query doesn't request it this way.

Please verify.

An easy alternative is just `SELECT transactions WHERE block_id = blabla` (This is over-simplification, but you get the point.)

Reply:
However, this is quite a problem regardless, so I'll be happy if you are able to fix it.